swarm/api/routes/db.py

```
@@ -242,6 +242,12 @@ async def get_db_stats():
 
         # Query counts from each table
         def safe_count(table: str) -> int:
+            # Explicit allowlist to prevent SQL injection via f-string
+            allowed_tables = {"runs", "steps", "tool_calls", "file_changes", "events", "facts"}
+            if table not in allowed_tables:
+                logger.warning(f"Attempted to query unauthorized table: {table}")
```

Ruff (0.15.9)

[warning] Logging statement uses f-string

(G004)